devops/ansible/roles/girder/templates/daemon/girder.service.j2

```diff
@@ -10,7 +10,19 @@ Restart=always
 {% if girder_virtualenv is defined %}
 ExecStart={{ girder_virtualenv }}/bin/python -m girder
 {% else %}
-ExecStart=python -m girder
+ExecStart={{ ansible_python.executable }} -m girder
+{% endif %}
+
+{% if girder_hardened %}
+# Hardened systemd options
+ProtectSystem=full
+ProtectHome=read-only
+ReadWriteDirectories={{ girder_path }} -{{ ansible_user_dir }}/.girder {{ girder_virtualenv|default("") }}
+
+#PrivateTmp=yes
+#PrivateDevices=yes
+
+# As of systemd >= 232, use PrivateUsers option and set ProtectSystem=strict
```

Comment:
User namespacing was added to systemd services in v232, and a more strict `ProtectSystem` setting was introduced which effectively turns the sandboxing into a whitelist instead of a blacklist.